### PR TITLE
Implementation of ResUNet architecture

### PR DESCRIPTION
**Pull request by hanaol**
This PR implements a ResUNet architecture as an alternative to the existing ResNet model. The original ResNet operates in the data dimension space, where an explicit initial downsampling step is required to accelerate training. In the ResNet/QM9 workflow, this was achieved by removing grid cells (pixels), which can lead to a loss of information.

In contrast, the proposed ResUNet architecture includes both encoder and decoder components, allowing it to handle downsampling and upsampling internally and more efficiently. Initial tests on the QM9 dataset indicate that the ResUNet achieves comparable performance and training speed to the ResNet/QM9 model. A more thorough comparison, however, will require further hyperparameter tuning for the ResUNet architecture.

### CODE REVIEW

File: src/electrai/model/resunet.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    return nn.Sequential(
+        nn.Conv3d(cin, cout, 3, stride=2, padding=1, padding_mode="circular"),
+        nn.InstanceNorm3d(cout),
+        nn.PReLU(),
+    )
+
+
+def upsample(cin, cout):
+    return nn.Sequential(
+        nn.ConvTranspose3d(cin, cout, kernel_size=2, stride=2),
```
Comment (forklady42) [inline src/electrai/model/resunet.py]: Not a blocker here but curious about how you're thinking about the repeating nature of materials when upsampling. I know circular padding doesn't have a clear interpretation for transposed convolutions, so this seems fine. Another option, potentially worth trying would be to upsample first and then do convolution with circular padding.
